web/client/utils/ImmutableUtils.js

```diff
@@ -82,6 +85,29 @@ export const arrayDelete = (path, condition, object) => {
     }
     return object;
 };
+
+export const normalizeGeomerty = (state, action) => {
```

Comment:
ImmutabilityUtils is a set of function that handles Immutability
please remove reducer related stuff from this utils file.

what I intended with my previous meeting was to reuse the set function exported  here
which I can see you are not doing 

I'll try to be clearer:
instead of doing 
```js
let newData = assign({}, action.features);
```
we can do
```js 
    let newCoords = [];
    if (action.features[0].geometry.type === 'Point') {
        newCoords = [normalizeLng(newData[0].geometry.coordinates[0]), newData[0].geometry.coordinates[1]];
        return set("tempFeatures", set("features[0].geometry.coordinates", newCoords, action), state);
        
        or         
       return assign({}, state, {tempFeatures: set("features[0].geometry.coordinates", newCoords, action)});
```
please double check this and do the same changes to the various cases
let me know if you need clarifications